delivery_carrier_label_batch/wizard/generate_labels.py

```diff
@@ -240,23 +250,38 @@ def action_generate_labels(self):
             labels = self._get_all_files(batch)
             labels_by_f_type = self._group_labels_by_file_type(labels)
             for f_type, labels in labels_by_f_type.items():
-                labels_bin = [
-                    codecs.decode(label, "base64") for label in labels if label
-                ]
-                filename = batch.name + '.' + f_type
-                filedata = self._concat_files(f_type, labels_bin)
-                if not filedata:
-                    # Merging of `f_type` not supported, so we cannot
-                    # create the attachment
-                    continue
-                data = {
-                    'name': filename,
-                    'res_id': batch.id,
-                    'res_model': 'stock.picking.batch',
-                    'datas': codecs.encode(filedata, 'base64'),
-                    'datas_fname': filename,
-                }
-                self.env['ir.attachment'].create(data)
+                if f_type == 'zpl2' and not zpl2_batch_merge:
+                    # We do not want to merge zpl2
+                    # because too big file can failed on zebra printers
+                    for label in labels:
+                        filename = "%s.%s" % (label['name'], f_type)
+                        data = {
+                            'name': filename,
+                            'res_id': batch.id,
+                            'res_model': 'stock.picking.batch',
+                            'datas': label,
+                            'datas_fname': filename,
+                        }
+                        self.env['ir.attachment'].create(data)
+                else:
+                    labels_bin = [
+                        codecs.decode(label, "base64") for label in labels if label
```

Comment:
This line is too long. Please install pre-commit as explained in https://odoo-community.org/blog/the-oca-blog-1/post/black-isort-pre-commit-97 then run

```
pre-commit run --files delivery_carrier_label_batch/models/* delivery_carrier_label_batch/wizard/* 
```
and commit any changed file. This should fix Travis.

Reply:
Thank you for the instructions. I tried your command but i got the error below 
`An error has occurred: InvalidConfigError: 
=====> .pre-commit-config.yaml does not exist
Check the log at /root/.cache/pre-commit/pre-commit.log`

I think i should have a .pre-commit-config.yaml in my branche

Reply:
my bad, pre-commit is not set up on 12.0 branch, only 13.0...